### DISCUSSION TITLE
Less number of slices in the training data.

### DISCUSSION THREAD

**Discussion by ramchandracheke** [Project-MONAI/MONAI#2991] [Q&A]
Hi,
I am training a model with input 128x128x64, However, few samples have less than 64 slices in the training. Can you please suggest how to handle this problem.
Thanks,
Ram

**Comment by ramchandracheke** [comment]
```python
train_transforms = Compose(
    [
        LoadImaged(keys=["image", "label"]),
        AddChanneld(keys=["image", "label"]),
        ScaleIntensityRanged(
            keys=["image"], a_min=-200, a_max=300,
            b_min=0.0, b_max=1.0, clip=True,
        ),
        SpatialPadd(keys=["image", "label"],spatial_size=(128,128,64),method= 'symmetric'),
        RandCropByPosNegLabeld(
            keys=["image", "label"],
            label_key="label",
            spatial_size=(128, 128, 64),
            pos=1,
            neg=1,
            num_samples=4,
            image_key="image",
            image_threshold=0,
        ),
        ToTensord(keys=["image", "label"]),
    ]
)
```

**Reply by rijobro** [reply]
Can you remove the `RandCropByPosNegLabeld` and confirm that all images are **at least** `(128,128,64)`? The documentation says:

> if a dimension of the input data size is bigger than the pad size, will not pad that dimension

**Reply by ramchandracheke** [reply]
Hey @rijobro,
Thanks for reply. Some of the CT scan data contains less than 64 slices. Before, I did not apply below transformation. 
```python 
SpatialPadd(keys=["image", "label"],spatial_size=(128,128,64),method= 'symmetric'),
```
But after adding this transformation it works. It is okay to add this order of transformation?
Thanks,
Ram

**Reply by rijobro** [reply]
Yeah, if you do `SpatialPadd` followed by `RandCropByPosNegLabeld`, then you're essentially saying "I want all of my images to be at least `(128,128,64)`, and then if any are bigger, I want you to take a random crop, such that the resulting image is `(128,128,64)`.
